### PR TITLE
merge heuristic-function-todo with develop

### PR DESCRIPTION
**Pull request by ericchase**
this update contains new code for the heuristic function and its helpers
it also includes completion of the Cell class and more test cases

### CODE REVIEW

File: code/src/cell.ts [hunk 1]
```diff
@@ -1,5 +1,15 @@
 export class Cell {
-  constructor(readonly location: number) {
-    if (location < 0 || location > 99) throw Error('location out of bounds');
+  readonly r: number;
```
Comment (CAReyes) [inline code/src/cell.ts]: Remove

File: code/src/cell.ts [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+  readonly c: number;
```
Comment (CAReyes) [inline code/src/cell.ts]: remove

File: code/src/cell.ts [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+
+  constructor(readonly row: number, readonly col: number) {
+    if (row < 0 || row > 9) throw Error('cell out of bounds');
+    if (col < 0 || col > 9) throw Error('cell out of bounds');
+    this.r = row;
```
Comment (CAReyes) [inline code/src/cell.ts]: remove

File: code/src/cell.ts [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
+    this.c = col;
```
Comment (CAReyes) [inline code/src/cell.ts]: remove

File: code/src/cell.ts [hunk 5]
```diff
@@ -1,5 +1,15 @@
...
   }
-}
\ No newline at end of file
+
+  rc() {
```
Comment (CAReyes) [inline code/src/cell.ts]:
```suggestion
  get linearPosition() => this.row * 10 + this.col;
```

File: code/src/cell.ts (hunk 1 above)
Comment (CAReyes) [inline code/src/cell.ts]: Remove

File: code/src/cell.ts (hunk 2 above)
Comment (CAReyes) [inline code/src/cell.ts]: remove

File: code/src/cell.ts (hunk 5 above)
Comment (CAReyes) [inline code/src/cell.ts]: refactor to 
```suggestion
  get linearPosition() => this.row * 10 + this.col;
```

File: code/src/heuristic-function.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+export function make_obstacle_list(...cells: Cell[]): boolean[] {
+  let obstacles: boolean[] = [];
+  for (let c of cells) {
+    obstacles[c.rc()] = true;
+  }
+  return obstacles;
+}
+
+export function manhattan_distance(a: Cell, b: Cell): number {
+  return Math.abs(a.r - b.r) + Math.abs(a.c - b.c);
```
Comment (CAReyes) [inline code/src/heuristic-function.ts]: r -> row
c -> col

File: code/src/cell.ts (hunk 5 above)
Comment (ericchase) [reply code/src/cell.ts]: couldn't figure out how to use lambda function for this

**Comment by ericchase** [comment]
uh, changes were made... now what

**Pull request merged**